### PR TITLE
dpkg: runtime fixes for perl and shasums

### LINKED ISSUE #38990: dpkg/1.17.25 broken - Can't locate Dpkg.pm in @INC, version 1.17.21 OK
I started seeing this using dpkg-scanpackages:
```
Can't locate Dpkg.pm in @INC (you may need to install the Dpkg module) (@INC contains: /Library/Perl/5.18/darwin-thread-multi-2level /Library/Perl/5.18 /Network/Library/Perl/5.18/darwin-thread-multi-2level /Network/Library/Perl/5.18 /Library/Perl/Updates/5.18.2 /System/Library/Perl/5.18/darwin-thread-multi-2level /System/Library/Perl/5.18 /System/Library/Perl/Extras/5.18/darwin-thread-multi-2level /System/Library/Perl/Extras/5.18 .) at /usr/local/Cellar/dpkg/1.17.25/bin/dpkg-scanpackages line 27.
BEGIN failed--compilation aborted at /usr/local/Cellar/dpkg/1.17.25/bin/dpkg-scanpackages line 27.
```
I tested it with the other version in had in the Cellar and that worked:
```
%> /usr/local/Cellar/dpkg/1.17.21/bin/dpkg-scanpackages binary binary/override | gzip -9c > binary/Packages.gz
...
info: Wrote 33 entries to output Packages file.
```
while explicitly specifying 1.17.25 does not:
```
%> /usr/local/Cellar/dpkg/1.17.25/bin/dpkg-scanpackages binary binary/override | gzip -9c > binary/Packages.gz
Can't locate Dpkg.pm in @INC (you may need to install the Dpkg module) (@INC contains: /Library/Perl/5.18/darwin-thread-multi-2level /Library/Perl/5.18 /Network/Library/Perl/5.18/darwin-thread-multi-2level /Network/Library/Perl/5.18 /Library/Perl/Updates/5.18.2 /System/Library/Perl/5.18/darwin-thread-multi-2level /System/Library/Perl/5.18 /System/Library/Perl/Extras/5.18/darwin-thread-multi-2level /System/Library/Perl/Extras/5.18 .) at /usr/local/Cellar/dpkg/1.17.25/bin/dpkg-scanpackages line 27.
BEGIN failed--compilation aborted at /usr/local/Cellar/dpkg/1.17.25/bin/dpkg-scanpackages line 27.
```

### PR DESCRIPTION
**Pull request by DomT4**
The whole show here is moving to libexec. The Perl problem we had several versions ago and patched turns out to exist still.

We previously put in a fairly dirty patch for this, essentially hacking out that element from the configure script. Given the continuing problem I've just dumped everything in libexec here and env scripted it.

The env script is way less likely to break long-term than the dirty patch we were using prior to the last release. My testing passed fine for the last version bump, but evidently not every element of dpkg calls for the Perl elements at run-time.

I also fixed a couple of rogue quotation marks on one of the inreplaces.

Closes #38990

### CODE REVIEW

File: Library/Formula/dpkg.rb [hunk 1]
```diff
@@ -19,16 +19,23 @@ def install
     inreplace "scripts/Dpkg/Checksums.pm" do |s|
       s.gsub! "md5sum", "md5"
       s.gsub! "sha1sum", "shasum"
-      s.gsub! "sha256sum", "'shasum', '-a', '256'"
+      s.gsub! "sha256sum", "shasum', '-a', '256"
```
Comment (dunn) [inline Library/Formula/dpkg.rb]: unbalanced quote marks `"shasum'` and `'256"`. should this maybe just be `s.gsub! "sha256sum", "shasum -a 256"`?

Comment (DomT4) [inline Library/Formula/dpkg.rb]: This is actually correct in this case. The removal of the quote marks here is deliberate.

The original code is: ` program => [ 'sha256sum' ], `, so when we inreplaced as I did originally here we managed to change that to ` program => [ ''shasum', '-a', '256''" ] ` which obviously confused the heck out of the perl-using executables at runtime. We could flip it the other way around, to include the single quotes inside the double quotes around `sha256sum` but I don't think it's any less or more clear one way or another.

As for the spacing - It matches the patch we sent upstream previously. I generally try not to send a patch upstream and then do something else for the sake of not confusing both upstream and us further down the line :smiley_cat:.

Comment (dunn) [inline Library/Formula/dpkg.rb]: Ohh, gotcha.

**Comment by mikemcquaid** [comment]
Hooray for avoiding patches! Thanks @DomT4!

**Pull request closed without merging**